content/blog/part-7-how-collaborative-teams-of-agents-unlock-new-intelligence.md

```diff
@@ -0,0 +1,194 @@
+name="Academic Paper Researcher",
+model=OpenAIChat("gpt-4o"),
+role="Research academic papers...",
+tools=\[GoogleSearchTools(), ArxivTools()],
+add_name_to_instructions=True,
+instructions=dedent("""You are an academic researcher..."""),
+)
+```
+
+## Twitter - X Agent
+
+```python
+twitter_researcher = Agent(
+name="Twitter Researcher",
+model=OpenAIChat("gpt-4o"),
+role="Research Twitter/X topics",
+tools=[DuckDuckGoTools()],
+add_name_to_instructions=True,
+instructions=dedent("""You are a Twitter researcher..."""),
+
+)
+```
+
+### 3. Define the Team
+
+```python
+agent_team = Team(
+name="Discussion Team",
+mode="collaborate",
+model=OpenAIChat("gpt-4o"),
+members=[
+     reddit_researcher,
+     hackernews_researcher,
+     academic_paper_researcher,
+     twitter_researcher,
+
+],
+
+instructions=[
+     "You are a discussion master.",
+     "You must conclude the discussion once consensus is reached.",
+],
+
+success_criteria="The team has reached a consensus.",
+update_team_context=True,
+send_team_context_to_members=True,
+show_tool_calls=True,
+markdown=True,
+show_members_responses=True,
+)
+```
+
+### 4. Running the Discussion
+
+```python
+if __name__ == "__main__":
+
+asyncio.run(
+     agent_team.print_response(
+         message="Start the discussion on the topic: 'What is the best way to learn to code?'",
+         stream=True,
+         stream_intermediate_steps=True,
+     )
+)
+```
+
+### Example output
+
+```python
+* Reddit: Focus on project-building and freeCodeCamp
+* HackerNews: Start with Python and open-source
+* Academia: Reinforce with spaced repetition and mentorship
+* Twitter/X: Emphasize consistency and public learning
+* Team Consensus: Use beginner-friendly languages, build real-world projects, and immerse yourself in learning communities.
+```
+
+<center><img src="/img/screenshot-2025-07-21-at-4.06.02 pm.png" width="600" height="550" alt="Agent Parameters" title="Agent Parameters"></center>
+
+## Pro Tip: Run Agents in parallel
+
+```python
+asyncio.run(
+
+agent_team.print_response(
+     message="Start the discussion on the topic: 'How should we improve remote team collaboration?'",
+     stream=True,
+     stream_intermediate_steps=True,
+))
+```
+
+Using asyncio ensures agents work simultaneously, which dramatically boosts speed and output quality—especially in research-heavy or time-sensitive use cases.
+
+## Final thought's
+
+Collaborate Mode is more than a clever orchestration pattern—it’s the embodiment of distributed intelligence. By mimicking the structure of human brainstorming, it allows AI to perform with greater breadth, depth, and creativity. With frameworks like AGNO making implementation seamless, the age of intelligent, agent-led collaboration is no longer speculative—it’s operational.
+
+> As we continue evolving from single-shot prompts to structured autonomy, Collaborate Mode stands out as a key innovation for scalable, multi-perspective problem-solving in AI systems.
```

Comment:
I don't think you really mean to indent this section... do you?

<img width="1653" height="577" alt="image" src="https://github.com/user-attachments/assets/6a567363-0936-4914-8bfd-712e14040240" />